airbyte-local-cli-nodejs/src/command.ts

```diff
@@ -0,0 +1,286 @@
+      }
+    }
+  });
+  logger.debug('Source Config:', srcConfig);
+  logger.debug('Destination Config:', dstConfig);
+  return {srcConfig, dstConfig};
+}
+
+// Convert the options to CliOptions type
+function convertToCliOptions(options: any) {
+  return {
+    ...options,
+    srcImage: options.src,
+    dstImage: options.dst,
+  } as CliOptions;
+}
+
+// Validate the input options
+function validateConfigFileInput(config: FarosConfig, inputType: AirbyteConfigInputType) {
+  if (!config.src?.image && !config.srcInputFile) {
+    if (inputType === AirbyteConfigInputType.OPTION) {
+      throw new Error(`Missing source image. Please use '--src <image>' to provide the source image`);
+    } else {
+      throw new Error('Missing source image. Please make sure you provide it in the config file.');
+    }
+  }
+  if (!config.dst?.image && !config.srcOutputFile) {
+    if (inputType === AirbyteConfigInputType.OPTION) {
+      throw new Error(`Missing destination image. Please use '--dst <image>' to provide the destination image`);
+    } else {
+      throw new Error('Missing destination image. Please make sure you provide it in the config file.');
+    }
+  }
+  if (!config.src?.config || !config.dst?.config) {
+    logger.warn('No source or destination config provided. Please make sure this is intended.');
+  }
+}
+
+// parse the command line arguments
+export async function parseAndValidateInputs(argv: string[]) {
+  // Parse the command line arguments
+  const program = command().parse(argv);
+
+  // Check for unknown options
+  const unknown = program.parseOptions(argv).unknown;
+  unknown.forEach((u) => {
+    if (u.startsWith('--') && !u.startsWith('--src.') && !u.startsWith('--dst.')) {
+      throw new Error(`Unknown option: ${u}`);
+    }
+  });
+
+  // Handle the src and dst config options passed as options
+  const configKeyValues = parseSrcAndDstConfig(argv);
+  program.setOptionValue('srcConfig', configKeyValues.srcConfig);
+  program.setOptionValue('dstConfig', configKeyValues.dstConfig);
+
+  // Get the options
+  const options = program.opts();
+  logger.debug({options}, 'Options');
+  const cliOptions = convertToCliOptions(options);
+  logger.debug({cliOptions}, 'Cli options');
+
+  // Convert the cli options to FarosConfig
+  const farosConfig: FarosConfig = {
+    // The default source output file is stdout if `srcOnly` is true
+    // Take the non-default value if provided with `srcOutputFile` option
+    srcOutputFile: cliOptions.srcOnly ? '/dev/null' : cliOptions.srcOutputFile,
+    // Rename the `dstOnly` file path to `srcInputFile`
+    srcInputFile: cliOptions.dstOnly,
+    srcCheckConnection: cliOptions.srcCheckConnection,
+    dstUseHostNetwork: cliOptions.dstUseHostNetwork,
+    // If `dstOnly` is true, do not pull the source image. Otherwise, fall back to option `noSrcPull`
+    srcPull: cliOptions.dstOnly ? false : cliOptions.srcPull,
+    dstPull: cliOptions.srcOnly ? false : cliOptions.dstPull,
```

Comment:
this has to do with commander auto parses `--no-src-pull` to `srcPull` instead of `noSrcPull`.